grepper_python/main.py

```diff
@@ -0,0 +1,152 @@
+"""
+The MIT License
+
+Copyright (c) 2010-2023 Grepper, Inc. (https://www.grepper.com)
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+"""
+
+from __future__ import annotations
+
+from typing import Optional
+
+from .answer import GrepperAnswer
+from .exceptions import *
+
+import requests
+
+
+def exception_handler(status_code: str):
+    if status_code == "400":
+        return BadRequest
+    elif status_code == "401":
+        return Unauthorized
+    elif status_code == "403":
+        return Forbidden
+    elif status_code == "404":
+        return NotFound
+    elif status_code == "405":
+        return MethodNotAllowed
+    elif status_code == "429":
+        return TooManyRequests
+    elif status_code == "500":
+        return InternalServerError
+    elif status_code == "503":
+        return ServiceUnavailable
+
+
+class Grepper:
+    """
+    Python Grepper API Wrapper
+    """
+
+    def __init__(self, __api_key: str):
+        self.__api_key = __api_key
```

Comment:
```suggestion
class Grepper:
    """
    Python Grepper API Wrapper
    """
    _api_key: str

    def __init__(self, api_key: str):
        self._api_key = _api_key
```

No need to use a double `_`. In Python, a private variable only needs a single leading underscore (Unless you need name mangling).

See https://www.python-engineer.com/posts/double-single-leading-underscore/ for more information

Reply:
Ahhh I didn't know that before thank you so much for pointing that out!